--- mplayer/mplayer.c.format~	2012-05-15 22:56:53.000000000 +0200
+++ mplayer/mplayer.c	2012-05-17 01:44:24.525344440 +0200
@@ -2918,7 +2918,7 @@ int main(int argc, char *argv[])
 
     if (!filename && !player_idle_mode && !use_gui) {
         // no file/vcd/dvd -> show HELP:
-        mp_msg(MSGT_CPLAYER, MSGL_INFO, help_text);
+        mp_msg(MSGT_CPLAYER, MSGL_INFO, "%s", help_text);
         exit_player_with_rc(EXIT_NONE, 0);
     }
 
