--- MPlayer-1.1.1/libdvdread4/dvd_input.c.orig	2013-05-09 10:06:43.871182500 +0200
+++ MPlayer-1.1.1/libdvdread4/dvd_input.c	2013-05-09 10:07:17.439762835 +0200
@@ -48,12 +48,7 @@ char *      (*dvdinput_error) (dvd_input
 #else
 
 /* dlopening libdvdcss */
-#ifdef HAVE_DLFCN_H
 #include <dlfcn.h>
-#else
-/* Only needed on MINGW at the moment */
-#include "../../msvc/contrib/dlfcn.c"
-#endif
 
 typedef struct dvdcss_s *dvdcss_t;
 static dvdcss_t (*DVDcss_open)  (const char *);
