--- libmpdemux/mp_taglists.h.orig	2012-10-23 17:23:52.431578688 +0400
+++ libmpdemux/mp_taglists.h	2012-10-23 17:23:58.225780029 +0400
@@ -20,6 +20,8 @@
 #define MPLAYER_MP_TAGLISTS_H
 
 #include <stdint.h>
+#include "libavutil/common.h"
+#include "libavformat/avformat.h"
 
 enum CodecID mp_tag2codec_id(uint32_t tag, int audio);
 uint32_t mp_codec_id2tag(enum CodecID codec_id, uint32_t old_tag, int audio);
