--- mplayer-1.0pre1/libvo/video_out.c.mga	2003-09-01 20:24:27.000000000 +0200
+++ mplayer-1.0pre1/libvo/video_out.c	2003-09-02 19:28:56.000000000 +0200
@@ -118,6 +118,13 @@
 #ifdef HAVE_DIRECTX
         &video_out_directx,
 #endif
+#ifdef HAVE_XV
+        &video_out_xv,
+#endif
+#ifdef HAVE_X11
+        &video_out_x11,
+        &video_out_xover,
+#endif
 #ifdef HAVE_XMGA
         &video_out_xmga,
 #endif
@@ -136,13 +143,6 @@
 #ifdef HAVE_XVMC
         &video_out_xvmc,
 #endif
-#ifdef HAVE_XV
-        &video_out_xv,
-#endif
-#ifdef HAVE_X11
-        &video_out_x11,
-        &video_out_xover,
-#endif
 #ifdef HAVE_GL
         &video_out_gl,
         &video_out_gl2,
