--- mplayer/libmpcodecs/vf_spp.c.ffmp~	2012-05-17 03:35:04.710553909 +0200
+++ mplayer/libmpcodecs/vf_spp.c	2012-05-17 03:35:08.653578743 +0200
@@ -579,7 +579,7 @@ static int vf_open(vf_instance_t *vf, ch
     init_avcodec();
 
     vf->priv->avctx= avcodec_alloc_context3(NULL);
-    ff_dsputil_init(&vf->priv->dsp, vf->priv->avctx);
+    dsputil_init(&vf->priv->dsp, vf->priv->avctx);
 
     vf->priv->log2_count= 3;
 
